task.py

```diff
@@ -43,6 +44,24 @@ def get_template_args(self) -> dict[str, str]:
             "node_name": self.node_name,
         }
 
+    def render_file(
+        self,
+        log_info: str,
+        in_file_template: Optional[str] = None,
```

Comment:
Do we really need to include these optional args? I don't see a case where they would be used. We could probably even derive the log info string from the Task itself rather than requiring it as an argument to make things even easier for the caller.